### PR TITLE
Sync with upstream

### PR DESCRIPTION
**Pull request by sheldongill**
Pull in changes from upstream, including our feature/ios-swift branch modification

### CODE REVIEW

File: fh-ios-sdk/FHConfig.m [hunk 1]
```diff
@@ -21,13 +21,18 @@ - (instancetype)init {
...
         } else {
-            @throw([NSException exceptionWithName:@"fhconfigException"
-                                           reason:@"fhconfig.plist was not located"
-                                         userInfo:nil]);
+            path = [[NSBundle bundleForClass:[self class]] pathForResource:@"fhconfig" ofType:@"plist"];
+            if (path) {
+                self.properties = [NSMutableDictionary dictionaryWithContentsOfFile:path];
+            } else {
+                @throw([NSException exceptionWithName:@"fhconfigException"
+                                               reason:@"fhconfig.plist was not located"
```
Comment (sheldongill) [inline fh-ios-sdk/FHConfig.m:33]: This is a rubbish reason message really. "Couldn't find required configuration file fhconfig.plist" would be better.

File: fh-ios-sdk/FHConfig.m [hunk 2]
```diff
@@ -21,13 +21,18 @@ - (instancetype)init {
...
         NSString *path =
-            [[NSBundle bundleForClass:[self class]] pathForResource:@"fhconfig" ofType:@"plist"];
+            [[NSBundle mainBundle] pathForResource:@"fhconfig" ofType:@"plist"];
         if (path) {
             self.properties = [NSMutableDictionary dictionaryWithContentsOfFile:path];
         } else {
-            @throw([NSException exceptionWithName:@"fhconfigException"
-                                           reason:@"fhconfig.plist was not located"
-                                         userInfo:nil]);
+            path = [[NSBundle bundleForClass:[self class]] pathForResource:@"fhconfig" ofType:@"plist"];
```
Comment (lordmortis) [inline fh-ios-sdk/FHConfig.m:28]: Uh what? Is it likely FHConfig is going to be in the framework bundle for fh-ios-sdk ? (or the cocoa pod framework bundle?)

File: fh-ios-sdk/FHConfig.m (hunk 1 above)
Comment (lordmortis) [inline fh-ios-sdk/FHConfig.m:33]: it should literally be "Could not find fhconfig.plist in the main application bundle or fh-ios-sdk bundle" 
probably with some directions to what should be in that file

File: fh-ios-sdk/FHConfig.m (hunk 2 above)
Comment (sheldongill) [inline fh-ios-sdk/FHConfig.m:28]: I thought maybe initially as you could have a "default" config which talks to your instance. A bit more thought and that seems dumb and unlikely. You need to spin up your app with an appropriate configuration and there is no "default" config which makes sense.

File: fh-ios-sdk/FHConfig.m (hunk 1 above)
Comment (sheldongill) [inline fh-ios-sdk/FHConfig.m:33]: "Couldn't find required configuration file fhconfig.plist in the main application bundle."

**Pull request merged**